File: model/src/main/scala/ru/mipt/acsl/decode/model/domain/impl/naming/ElementNameImpl.scala
package ru.mipt.acsl.decode.model.domain.impl.naming

import ru.mipt.acsl.decode.model.domain.pure.{naming => n}

/**
  * @author Artem Shein
  */
private case class ElementNameImpl(value: String) extends n.ElementName {
  override def asMangledString: String = value
}

File: model/src/main/java/ru/mipt/acsl/decode/model/naming/ElementNameImpl.java
package ru.mipt.acsl.decode.model.naming;

import java.util.regex.Pattern;

/**
 * Created by metadeus on 06.06.16.
 */
public class ElementNameImpl implements ElementName {

    private final String mangledName;

    @Override
    public String mangledNameString() {
        return mangledName;
    }

    ElementNameImpl(String mangledName) {
        this.mangledName = mangledName;
    }
}
